File: BackEnd/DotNet/WebDigital/WebDigital/Controllers/CartController.cs
using System.IdentityModel.Tokens.Jwt;
using System.Security.Claims;
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Mvc;
using WebDigital.dto;
using WebDigital.Services.Interface;

namespace WebDigital.Controllers
{
    [ApiController]
    [Route("api/cart")]
    [Authorize] // any logged-in user (not admin-only)
    public class CartController : ControllerBase
    {
        private readonly ICartService _service;

        public CartController(ICartService service)
        {
            _service = service;
        }

        // Pulls the current user's id out of the JWT — never trust a client-sent user id
        private int GetCurrentUserId()
        {
            var userIdClaim = User.FindFirst(JwtRegisteredClaimNames.Sub)
                ?? User.FindFirst(ClaimTypes.NameIdentifier);

            return int.Parse(userIdClaim!.Value);
        }

        [HttpGet]
        public async Task<IActionResult> GetCart()
        {
            var userId = GetCurrentUserId();
            var cart = await _service.GetCart(userId);
            return Ok(cart);
        }

        [HttpPost]
        public async Task<IActionResult> AddToCart(CartAddDto dto)
        {
            var userId = GetCurrentUserId();
            var cartId = await _service.AddToCart(userId, dto);
            return Ok(new { cartId });
        }

        [HttpPut("{cartId}")]
        public async Task<IActionResult> UpdateQuantity(int cartId, CartUpdateDto dto)
        {
            var userId = GetCurrentUserId();
            await _service.UpdateQuantity(userId, cartId, dto);
            return Ok("Cart item updated");
        }

        [HttpDelete("{cartId}")]
        public async Task<IActionResult> RemoveItem(int cartId)
        {
            var userId = GetCurrentUserId();
            await _service.RemoveItem(userId, cartId);
            return Ok("Cart item removed");
        }

        [HttpDelete]
        public async Task<IActionResult> ClearCart()
        {
            var userId = GetCurrentUserId();
            await _service.ClearCart(userId);
            return Ok("Cart cleared");
        }
    }
}

File: BackEnd/Java/transaction-service/src/main/java/com/digitalshelf/transactionservice/controller/CartController.java
package com.digitalshelf.transactionservice.controller;

import com.digitalshelf.transactionservice.dto.CartAddDto;
import com.digitalshelf.transactionservice.dto.CartItemDto;
import com.digitalshelf.transactionservice.dto.CartUpdateDto;
import com.digitalshelf.transactionservice.security.CurrentUserUtil;
import com.digitalshelf.transactionservice.service.CartService;
import lombok.RequiredArgsConstructor;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.List;
import java.util.Map;

@RestController
@RequestMapping("/api/cart")
@RequiredArgsConstructor
public class CartController {

    private final CartService cartService;
    private final CurrentUserUtil currentUserUtil;

    @GetMapping
    public ResponseEntity<List<CartItemDto>> getCart() {
        return ResponseEntity.ok(cartService.getCart(currentUserUtil.getCurrentUserId()));
    }

    @PostMapping
    public ResponseEntity<Map<String, Integer>> addToCart(@RequestBody CartAddDto dto) {
        Integer cartId = cartService.addToCart(currentUserUtil.getCurrentUserId(), dto);
        return ResponseEntity.ok(Map.of("cartId", cartId));
    }

    @PutMapping("/{cartId}")
    public ResponseEntity<String> updateQuantity(@PathVariable Integer cartId, @RequestBody CartUpdateDto dto) {
        cartService.updateQuantity(currentUserUtil.getCurrentUserId(), cartId, dto);
        return ResponseEntity.ok("Cart item updated");
    }

    @DeleteMapping("/{cartId}")
    public ResponseEntity<String> removeItem(@PathVariable Integer cartId) {
        cartService.removeItem(currentUserUtil.getCurrentUserId(), cartId);
        return ResponseEntity.ok("Cart item removed");
    }

    @DeleteMapping
    public ResponseEntity<String> clearCart() {
        cartService.clearCart(currentUserUtil.getCurrentUserId());
        return ResponseEntity.ok("Cart cleared");
    }
}